service/webhooks_service.go

```diff
@@ -0,0 +1,100 @@
+package service
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/libsv/bitcoin-hc/domains"
+	"github.com/libsv/bitcoin-hc/repository"
+	"github.com/spf13/viper"
+)
+
+// WebhookMaxTries is the maximum number of times a webhook will be retried.
+const WebhookMaxTries = "webhook.maxTries"
+
+// WebhooksService represents Webhooks service and provide access to repositories.
+type WebhooksService struct {
+	repo *repository.Repositories
+}
+
+// GenerateWebhook generates and save new webhook.
+func (s *WebhooksService) GenerateWebhook(url, tHeader, token string) (*domains.Webhook, error) {
+	webhook := domains.CreateWebhook(url, tHeader, token)
+	err := s.repo.Webhooks.AddWebhookToDatabase(webhook)
+	if err != nil {
+		w, _ := s.repo.Webhooks.GetWebhookByUrl(url)
```

Comment:
if there will be any database issue we will return invalid error info that 
`"webhook already exists and is active"`.

IMHO we should handle the error here first